script/MangroveDeployer.s.sol

```diff
@@ -20,32 +20,40 @@ contract MangroveDeployer is Deployer {
     innerRun({
       chief: envHas("CHIEF") ? vm.envAddress("CHIEF") : broadcaster(),
       gasprice: envHas("GASPRICE") ? vm.envUint("GASPRICE") : 1,
-      gasmax: envHas("GASMAX") ? vm.envUint("GASMAX") : 2_000_000
+      gasmax: envHas("GASMAX") ? vm.envUint("GASMAX") : 2_000_000,
+      gasbot: envHas("GASBOT") ? vm.envAddress("GASBOT") : fork.get("Gasbot")
     });
     outputDeployment();
   }
 
-  function innerRun(address chief, uint gasprice, uint gasmax) public {
+  function innerRun(address chief, uint gasprice, uint gasmax, address gasbot) public {
     broadcast();
     if (forMultisig) {
-      mgv = new Mangrove{salt:salt}({governance: chief, gasprice: gasprice, gasmax: gasmax});
+      oracle = new MgvOracle{salt: salt}({governance_: chief, initialMutator_: gasbot});
     } else {
-      mgv = new Mangrove({governance: chief, gasprice: gasprice, gasmax: gasmax});
+      oracle = new MgvOracle({governance_: chief, initialMutator_: gasbot});
+    }
+    fork.set("MgvOracle", address(oracle));
+
+    broadcast();
+    if (forMultisig) {
+      mgv = new Mangrove{salt:salt}({governance: broadcaster(), gasprice: gasprice, gasmax: gasmax});
+    } else {
+      mgv = new Mangrove({governance: broadcaster(), gasprice: gasprice, gasmax: gasmax});
     }
     fork.set("Mangrove", address(mgv));
 
+    broadcast();
+    mgv.setMonitor(address(oracle));
+    broadcast();
+    mgv.setUseOracle(true);
```

Comment:
This means we'll always use an oracle for instance also for instance for MangroveJsDeploy where I do not see it being used today. Is that intended? Maybe we have JS-tests that will break if they try to set gasprices

Reply:
Good point. The intention was to have one script that deploys & configures all core&periphery contracts (Mangrove, MgvReader, MgvOracle, and MangroveOrder (tbd)) in a default setup across environments.

There are indeed JS-tests that set the gasprice.

Since we're planning to use a gas oracle in all environments, I'd suggest we adapt those JS tests to set the gasprice via the oracle.